sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/DeduplicateRelations.scala

```diff
@@ -220,7 +220,52 @@ object DeduplicateRelations extends Rule[LogicalPlan] {
           if (attrMap.isEmpty) {
             planWithNewChildren
           } else {
-            planWithNewChildren.rewriteAttrs(attrMap)
+            def rewriteAttrsMatchWithSubPlan(
+                attrs: Seq[Attribute],
+                attrMap: AttributeMap[Attribute],
+                planOutput: Seq[Attribute]): Seq[Attribute] = {
+              val canRewriteAttrs = attrMap.filter(a => planOutput.contains(a._2))
+              attrs.map(attr => canRewriteAttrs.getOrElse(attr, attr))
+            }
+
+            def rewriteOrderMatchWithSubPlan(
+                attrs: Seq[SortOrder],
+                attrMap: AttributeMap[Attribute],
+                planOutput: Seq[Attribute]): Seq[SortOrder] = {
+              val canRewriteAttrs = attrMap.filter(a => planOutput.contains(a._2))
+              attrs.filter(_.child.isInstanceOf[Attribute]).map(attr => {
+                if (canRewriteAttrs.contains(attr.child.asInstanceOf[Attribute])) {
+                  attr.copy(child = canRewriteAttrs(attr.child.asInstanceOf[Attribute]))
+                } else {
+                  attr
+                }
+              })
+            }
+
+            planWithNewChildren match {
+              case c @ CoGroup(_, keyDeserializer, leftDeserializer, rightDeserializer,
+              leftGroup, rightGroup, leftAttr, rightAttr, leftOrder, rightOrder, _, left,
```

Comment:
nit: let's not use pattern match if there are too many parameters. `case c: CoGroup` is better.